### PR TITLE
feat(status footer): create footer with API status

### PR DESCRIPTION
**Pull request by elysee15**
Capture d’écran 2021-08-05 à 15 30 42
Capture d’écran 2021-08-05 à 15 32 15

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -22,10 +22,6 @@
         "@typescript-eslint"
     ],
     "rules": {
-        "indent": [
-            "error",
-            "tab"
-        ],
```
Comment (bbengfort) [inline .eslintrc.json:28]: We should create a ticket for linting as part of GitHub actions when we get to CI/CD.

File: src/components/Badge.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	}
+});
+
+type BadgeProps = {
+	color: string;
+	content: string;
+};
+
+const Badge: React.FC<BadgeProps> = props => {
+	const classes = useStyles(props);
```
Comment (bbengfort) [inline src/components/Badge.tsx:20]: Very cool use of pure CSS to make the circle icon!

File: src/components/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+						break;
+					case Status.maintainance:
+						setStatusColor("yellow");
+						break;
+				}
+			})
+			.catch((error: AxiosError) => {
+				console.error("[Footer] fetch server status", error.message);
+			});
+	}, []);
```
Comment (bbengfort) [inline src/components/Footer.tsx:42]: If you could remind me - `useEffect` with the `[]` second argument as a dependency means that this effect will only be run when the component is first loaded, not on every render, right?

File: src/components/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+				switch (response.data.status) {
+					case Status.ok:
+						setStatusColor("green");
+						break;
+					case Status.unhealthy:
+						setStatusColor("red");
+						break;
+					case Status.maintainance:
+						setStatusColor("yellow");
+						break;
```
Comment (bbengfort) [inline src/components/Footer.tsx:32]: Are these color names picking up the material-UI color palette or do they use the standard named css colors?

File: src/components/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+				}
+			})
+			.catch((error: AxiosError) => {
+				console.error("[Footer] fetch server status", error.message);
+			});
+	}, []);
+
+	return (
+		<footer className={classes.root}>
+			<Typography>Made with &spades; by Rotational Labs</Typography>
```
Comment (bbengfort) [inline src/components/Footer.tsx]: Could we please add a link to Rotational?

```
<Typography>Made with &spades; by <a href="https://rotational.io" target="_blank">Rotational Labs</a></Typography>
```

File: src/components/Footer.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+			<Box display="flex" alignItems="center" gridGap=".5rem">
+				<Badge color={statusColor} content="status" />
+				<Typography variant="caption">version: {serverStatus?.version}</Typography>
```
Comment (bbengfort) [inline src/components/Footer.tsx]: The margin on the version text is larger than the margin between "status" and the badge; could we balance them out so the "status" and "version" text are the same distance from the green circle?

File: src/services/status.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import { AxiosPromise } from "axios";
+import client from "config";
+
+export default function getStatus(config = {}): AxiosPromise {
+	return client.get("/status", config);
```
Comment (bbengfort) [inline src/services/status.ts:5]: Glad Axios is working now with the CORS configuration!

**Review by bbengfort — APPROVED** [review]
Looks great - excellent work! I've made a few comments and requested a few changes, feel free to squash and merge after you've responded to them!

File: src/components/Footer.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+			.catch((error: AxiosError) => {
+				console.error("[Footer] fetch server status", error.message);
+			});
+	}, []);
+
+	return (
+		<footer className={classes.root}>
+			<Typography>Made with &spades; by Rotational Labs</Typography>
+			<Box display="flex" alignItems="center" gridGap=".5rem">
+				<Badge color={statusColor} content="status" />
```
Comment (bbengfort) [inline src/components/Footer.tsx]: From our slack conversation, I was thinking something like:

```
<Tooltip title=`connected to ${endpoint}`>
  <Badge color={statusColor} content="status" />
  <Typography variant="caption">version: {serverStatus?.version}</Typography>
</Tooltip>
```

File: .eslintrc.json (hunk 1 above)
Comment (elysee15) [reply .eslintrc.json:28]: Alright

File: src/components/Footer.tsx (hunk 3 above)
Comment (elysee15) [reply src/components/Footer.tsx:42]: Yes, that's right

File: src/components/Footer.tsx (hunk 4 above)
Comment (elysee15) [reply src/components/Footer.tsx:32]: I do use css colors but I think I'll be changing it to material colors palette

File: src/components/Footer.tsx (hunk 5 above)
Comment (elysee15) [reply src/components/Footer.tsx]: Alright

File: src/components/Footer.tsx (hunk 6 above)
Comment (elysee15) [reply src/components/Footer.tsx]: Okay

File: src/components/Footer.tsx (hunk 8 above)
Comment (elysee15) [reply src/components/Footer.tsx]: Okay

File: src/components/Footer.tsx (hunk 3 above)
Comment (bbengfort) [reply src/components/Footer.tsx:42]: Perfect!

**Pull request merged**